File: app/src/main/java/com/morrowbone/mafiacards/app/adapter/DeckAdapter.java
package com.morrowbone.mafiacards.app.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.morrowbone.mafiacards.app.R;
import com.morrowbone.mafiacards.app.model.Deck;
import com.morrowbone.mafiacards.app.views.CardView;

public class DeckAdapter extends RecyclerView.Adapter<DeckAdapter.ViewHolder> {
    private Deck mDeck;

    public DeckAdapter(Deck deck) {
        mDeck = deck;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater layoutInflater = LayoutInflater.from(parent.getContext());
        View view = layoutInflater.inflate(R.layout.view_deck_card, null);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        holder.cardView.setPlayerNum(position + 1);
    }

    @Override
    public int getItemCount() {
        return mDeck.size();
    }

    static class ViewHolder extends RecyclerView.ViewHolder {
        CardView cardView;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            cardView = itemView.findViewById(R.id.cardView);
            cardView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {
                    cardView.flipit();
                }
            });
        }
    }
}

File: app/src/main/java/com/morrowbone/mafiacards/app/adapter/DeckAdapter.kt
package com.morrowbone.mafiacards.app.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView

import com.morrowbone.mafiacards.app.R
import com.morrowbone.mafiacards.app.data.Deck
import com.morrowbone.mafiacards.app.views.CardView

class DeckAdapter(private val mDeck: Deck) : RecyclerView.Adapter<DeckAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val layoutInflater = LayoutInflater.from(parent.context)
        val view = layoutInflater.inflate(R.layout.view_deck_card, null)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val card = mDeck.getCards().get(position)
        val cardView = holder.cardView
        cardView.setPlayerNum(position + 1)
        cardView.setCardImageResource(card.getImageResId())
        cardView.setRoleName(card.getTitle())
    }

    override fun getItemCount(): Int {
        return mDeck.getCards().size
    }

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var cardView: CardView

        init {
            cardView = itemView.findViewById(R.id.cardView)
            cardView.setOnClickListener { cardView.flipit() }
            cardView.setOnLongClickListener {
                cardView.flipit()
                true
            }
        }
    }
}
